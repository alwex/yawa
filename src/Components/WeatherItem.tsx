import React from 'react'
import { TouchableOpacity } from 'react-native'
import { Badge, Icon, Text, View } from 'native-base'
import toDate from 'date-fns/toDate'
import format from 'date-fns/format'
import style from './Styles/WeatherItemStyle'
import { translateIcon } from '../Lib/Helpers'

interface WeatherItemProps {
  onPress: () => void
  date: string
  temp: number
  humidity: number
  description: string
  iconCode: string
  locationName: string
}

export default class WeatherItem extends React.Component<WeatherItemProps> {
  render() {
    const { date, temp, description, iconCode, locationName, humidity, onPress } = this.props
    const parsedDate = toDate(date)
    const dayOfYear = format(parsedDate, 'dd', { awareOfUnicodeTokens: true })
    const dayOfWeek = format(parsedDate, 'iii', { awareOfUnicodeTokens: true })
    const time = format(parsedDate, 'p', { awareOfUnicodeTokens: true })
    return (
      <TouchableOpacity style={style.container} onPress={onPress}>
        <View style={style.mainRow}>
          <View>
            <View>
              <Text style={style.temp}>{temp}Â°</Text>
            </View>
            <View>
              <Text style={style.description}>{description.toUpperCase()}</Text>
            </View>
            <View>
              <Text style={style.status}>{locationName.toUpperCase()}</Text>
            </View>
          </View>
          <View>
            <Icon style={style.icon} type="MaterialCommunityIcons" name={translateIcon(iconCode)} />
          </View>
          <View>
            <View>
              <Badge style={style.humidityBadge}>
                <Icon
                  type="MaterialCommunityIcons"
                  name="water"
                  style={style.humidityBadgeContent}
                />
                <Text>{humidity}%</Text>
              </Badge>
            </View>
            <View>
              <Text style={style.dayOfYear}>{dayOfYear}</Text>
              <Text style={style.dayOfWeek}>{dayOfWeek.toUpperCase()}</Text>
              <Text style={style.time}>{time}</Text>
            </View>
          </View>
        </View>
      </TouchableOpacity>
    )
  }
}
