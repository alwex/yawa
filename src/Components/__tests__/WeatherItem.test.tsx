import React from 'react'
import renderer from 'react-test-renderer'
import WeatherItem from '../WeatherItem'

describe('WeatherItem', () => {
  it('render correctly', () => {
    const component = renderer
      .create(
        <WeatherItem
          temp={15.2}
          humidity={60.5}
          onPress={() => {}}
          locationName="Dunedin"
          date="2018-12-25 05:05:06"
          description="few shower"
          iconCode="50d"
        />
      )
      .toJSON()
    expect(component).toMatchSnapshot()
  })
})
