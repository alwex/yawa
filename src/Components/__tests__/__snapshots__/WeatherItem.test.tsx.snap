// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherItem render correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 1,
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {},
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-around",
        },
      ]
    }
  >
    <View
      style={Object {}}
    >
      <View
        style={Object {}}
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              },
              Object {
                "color": "#2C3A3F",
                "fontSize": 50,
              },
            ]
          }
          uppercase={false}
        >
          15.2
          °
        </Text>
      </View>
      <View
        style={Object {}}
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              },
              Object {
                "color": "#D6637A",
                "fontSize": 12,
                "fontWeight": "bold",
              },
            ]
          }
          uppercase={false}
        >
          FEW SHOWER
        </Text>
      </View>
      <View
        style={Object {}}
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              },
              Object {
                "color": "#90C7D4",
                "fontSize": 12,
                "fontWeight": "bold",
              },
            ]
          }
          uppercase={false}
        >
          DUNEDIN
        </Text>
      </View>
    </View>
    <View
      style={Object {}}
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Array [
              Object {
                "color": "#000",
                "fontSize": 30,
              },
              Object {
                "color": "#000",
                "fontSize": 30,
              },
              Object {
                "color": "#D6637A",
                "fontSize": 80,
              },
            ],
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
        type="MaterialCommunityIcons"
      >
        
      </Text>
    </View>
    <View
      style={Object {}}
    >
      <View
        style={Object {}}
      >
        <View
          style={
            Array [
              Object {
                "alignSelf": "flex-start",
                "backgroundColor": "#ED1727",
                "borderRadius": 13.5,
                "height": 27,
                "justifyContent": "center",
                "paddingBottom": 3,
                "paddingLeft": 6,
                "paddingRight": 6,
                "paddingTop": 3,
              },
              Object {
                "alignSelf": "center",
                "backgroundColor": "#D6637A",
                "flexDirection": "row",
              },
            ]
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Array [
                  Object {
                    "color": "#000",
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#000",
                    "fontSize": 30,
                  },
                  Object {
                    "alignSelf": "center",
                    "color": "white",
                    "fontSize": 14,
                  },
                ],
                Object {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
            type="MaterialCommunityIcons"
          >
            
          </Text>
          <Text
            style={
              Object {
                "color": "#fff",
                "fontFamily": "System",
                "fontSize": 15,
                "lineHeight": 19,
                "paddingLeft": 3,
                "paddingRight": 3,
                "textAlign": "center",
              }
            }
            uppercase={false}
          >
            60.5
            %
          </Text>
        </View>
      </View>
      <View
        style={Object {}}
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              },
              Object {
                "alignSelf": "center",
                "color": "#C9E4EB",
                "fontSize": 50,
                "fontWeight": "bold",
              },
            ]
          }
          uppercase={false}
        >
          25
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              },
              Object {
                "alignSelf": "center",
                "color": "#90C7D4",
                "fontSize": 30,
                "fontWeight": "bold",
              },
            ]
          }
          uppercase={false}
        >
          TUE
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              },
              Object {
                "alignSelf": "center",
                "color": "#2C3A3F",
                "fontSize": 12,
                "fontWeight": "normal",
              },
            ]
          }
          uppercase={false}
        >
          6:05 PM
        </Text>
      </View>
    </View>
  </View>
</View>
`;
